bigquery/storage/managedwriter/integration_test.go

```diff
@@ -513,6 +517,85 @@ func testInstrumentation(ctx context.Context, t *testing.T, mwClient *Client, bq
 	}
 }
 
+func testSchemaEvolution(ctx context.Context, t *testing.T, mwClient *Client, bqClient *bigquery.Client, dataset *bigquery.Dataset) {
+	testTable := dataset.Table(tableIDs.New())
+	if err := testTable.Create(ctx, &bigquery.TableMetadata{Schema: testdata.SimpleMessageSchema}); err != nil {
+		t.Fatalf("failed to create test table %s: %v", testTable.FullyQualifiedName(), err)
+	}
+
+	m := &testdata.SimpleMessageProto2{}
+	descriptorProto := protodesc.ToDescriptorProto(m.ProtoReflect().Descriptor())
+
+	// setup a new stream.
+	ms, err := mwClient.NewManagedStream(ctx,
+		WithDestinationTable(fmt.Sprintf("projects/%s/datasets/%s/tables/%s", testTable.ProjectID, testTable.DatasetID, testTable.TableID)),
```

Comment:
Interesting. So the users don't see the stream ID, either? I guess that gives us enough flexibility for the current schema evolution workaround.

Reply:
There's effectively two paths to getting a managed stream: allow `NewManagedStream()` to deal with the stream construction by specifying table/type/etc, or do it yourself and pass it in via `WithStreamName()` option.

Reply:
I expect users who explicitly do stream construction to be more likely to be doing dynamic proto schema stuff, as one of the things you get back from stream creation is table schema.

Reply:
Two reasons:  this is a bit of a standard in the APIs (see aip.dev for more context), and we want to avoid some circular dependencies by having the managedwriter depend on cloud.google.com/go/bigquery directly. 

https://github.com/googleapis/google-cloud-go/pull/5017 will make it easier to generate the string for the table resources if you end up using this option from a bigquery resource.